### ISSUE TITLE
call syntax for worker

### ISSUE THREAD

**Issue by ahabhgk** [web-infra-dev/rspack#3686]
create worker from function call syntax like `navigator.serviceWorker.register()`

**Comment by mszkwsk** [comment]
Hey! 

We are currently evaluating rspack as a replacement for Webpack. We were able to reduce build time from 20 to 1 second, thanks to this replacement. 

However, the output is not a functional application yet. We are using Service Worker in the application and currently there is no option for rspack to figure it out and load it like Webpack did. We would love to see that feature implemented, because it will ease the potential migration process on our end.

tl;dr bump, we are interested in this option:)

**Comment by karpiuMG** [comment]
@ahabhgk any plans to support it? what's the biggest issue there to have it? is there any other way to register a service worker?

**Comment by karpiuMG** [comment]
@ahabhgk are you able to provide any rough timeline? I'm sure you have a lot on yours plate right now, but any date would be appreciated to plan our work as well.

**Comment by ahabhgk** [comment]
Maybe next week, I'm working on it

**Comment by karpiuMG** [comment]
hooray! I am glad you've managed to implement that. When do you plan to release it? @ahabhgk

**Comment by ahabhgk** [comment]
If everything goes well, we will release this in v1.0.0-alpha.0 in tomorrow, it contains some break changes, but I think it should be easy to upgrade :)
